util/helm/helm.go

```diff
@@ -56,7 +56,7 @@ func IsMissingDependencyErr(err error) bool {
 	return strings.Contains(err.Error(), "found in requirements.yaml, but missing in charts")
 }
 
-func (h *helm) Template(appName string, namespace string, opts *argoappv1.ApplicationSourceHelm) ([]*unstructured.Unstructured, error) {
+func (h *helm) Template(appName string, namespace string, opts *argoappv1.ApplicationSourceHelm, commitSHA string) ([]*unstructured.Unstructured, error) {
```

Comment:
Can I ask:

* Rename `commitSHA` to`resolvedRevision`.
* Make it the third parameter.